x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/upgrade_prebuilt_rules_table/upgrade_rule_source_filter_popover.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useMemo } from 'react';
+import type { EuiSelectableOption } from '@elastic/eui';
+import { EuiFilterButton, EuiPopover, EuiSelectable } from '@elastic/eui';
+import { RuleSourceTypesEnum } from '../../../../rule_management/logic';
+import * as i18n from '../../../../../detections/pages/detection_engine/rules/translations';
+import { toggleSelectedGroup } from '../../../../../common/components/ml_popover/jobs_table/filters/toggle_selected_group';
+
+interface RuleSourceFilterPopoverProps {
+  selectedRuleSource: RuleSourceTypesEnum[];
+  onSelectedRuleSourceChanged: (newRuleSource: RuleSourceTypesEnum[]) => void;
+}
+
+const RULE_SOURCE_POPOVER_WIDTH = 200;
+
+const RuleSourceFilterPopoverComponent = ({
```

Comment:
`RuleSourceFilterPopoverComponent` -> `RuleCustomizationFilterPopoverComponent`. 

Please also update other occurrences of `RuleSource` to `RuleCustomization` within this file. This will future-proof the component, as we may introduce a table filter to differentiate external rules from internal ones. Distinguishing rule customization status from rule source type will help with that.